src/excel_loader.py

```diff
@@ -0,0 +1,236 @@
+
+
+def append_empty_rows(sheet, num_rows):
+    for _ in range(num_rows):
+        empty_row = [None] * sheet.max_column
+        sheet.append(empty_row)
+
+
+def copy_schedule_days(input_files, output_file_directory):
+    # Create a new workbook for the output if it doesn't exist
+    output_file_name = os.path.join(output_file_directory, "Daily_Schedules.xlsx")
+    if not os.path.exists(output_file_name):
+        output_wb = Workbook()
+        output_wb.save(output_file_name)
+    else:
+        output_wb = openpyxl.load_workbook(output_file_name, values_only=True)
+    
+    i = 0
+
+    for input_file in input_files:
+
+        df = pd.read_excel(input_file)
+
+        start_token = "Day " + str(input_schedule_days[i])
+        end_token = "Workbook"
+
+        start_index = find_token_row_index(start_token, df)
+        end_index = find_token_row_index(end_token, df)
+
+        if not start_index.empty and not end_index.empty:
+            start_index = start_index[0]
+            
+            # Find the first occurrence of the ending token after the start token
+            end_index = end_index[end_index > start_index][0]
+
+            # Extract the section of data between the tokens (excluding the tokens themselves)
+            selected_data = df.loc[start_index - 4: end_index, df.columns != df.columns[0]]  # Exclude the first column from being copied
+
+            # Get the selected sheet to copy data
+            sheet = output_wb.active
+
+            # Append the new data to the end of the sheet
+            for row in dataframe_to_rows(selected_data, index=False, header=False):
+                sheet.append(row)
+
+            # Define a custom style for datetime values
+            time_style = NamedStyle(name="time_style", number_format="HH:MM")
+
+            # Formatting date time
+            for row in sheet.iter_rows(min_row=sheet.max_row - len(selected_data) + 1, max_row=sheet.max_row):
+                for cell in row:
+                    source_cell = df.iloc[cell.row - sheet.max_row + len(selected_data) - 1, cell.column - 1]
+
+                     # Check if the value is a datetime object and apply the custom style
+                    if isinstance(source_cell, pd.Timestamp):
+                        cell.style = time_style
+                    
+                    #preserve the row height
+                    sheet.row_dimensions[cell.row].height = sheet.row_dimensions[cell.row - sheet.max_row + len(selected_data)].height
+
+            # Preserve column width
+            for column_cells in sheet.columns:
+                length = max(len(str(cell.value)) for cell in column_cells)
+                sheet.column_dimensions[column_cells[0].column_letter].width = length + 2  # Additional padding
+
+            output_wb.save(output_file_name)
+
+        else:
+            print("Day not found in study schedule")
+        
+        append_empty_rows(sheet, 4)
+
+        i=i+1
```

Comment:
I don't think this ever gets revisited, you might need to put this all in another big for loop or something

Reply:
it does, still adds the empty rows between schedules?